### PR TITLE
 allow failing for categorical and numerical output

### PR DESCRIPTION
**Pull request by maximlopin**
https://github.com/mindsdb/mindsdb_native/issues/106
- Model training can fail (which means an error will be logged and you will get an exception on attempt to call `predict()` on the `Predictor` instance) for numerical and categorical data types for now.
- Added `force_predict` boolean flag (`False` by default) to `advanced_args` which will allow to call `predict()` on the `Predictor` instance if model training failed.

### CODE REVIEW

**Comment by ZoranPandovski** [comment]
**CLA Assistant Lite** All Contributors have signed the CLA.

**Comment by ZoranPandovski** [comment]
**CLA Assistant Lite** All Contributors have signed the CLA.

**Comment by ZoranPandovski** [comment]
**CLA Assistant Lite** All Contributors have signed the CLA.

**Comment by ZoranPandovski** [comment]
**CLA Assistant Lite** All Contributors have signed the CLA.

**Comment by ZoranPandovski** [comment]
**CLA Assistant Lite** All Contributors have signed the CLA.

**Comment by ZoranPandovski** [comment]
**CLA Assistant Lite** All Contributors have signed the CLA.

**Comment by ZoranPandovski** [comment]
**CLA Assistant Lite** All Contributors have signed the CLA.

File: mindsdb_native/libs/phases/model_analyzer/model_analyzer.py [hunk 1]
```diff
@@ -29,6 +31,35 @@ def run(self):
...
+                if data_subtype == DATA_SUBTYPES.TAGS:
+                    # need to implement stats_v2[col]['guess_probability'] for tags
+                    pass
+                else:
+                    if accuracy_score(reals, preds) < self.transaction.lmd['stats_v2'][col]['guess_probability']:
+                        fails = True
+            elif data_type == DATA_TYPES.NUMERIC:
+                if r2_score(reals, preds) < 0:
+                    fails = True
+            else:
```
Comment (paxcema) [inline mindsdb_native/libs/phases/model_analyzer/model_analyzer.py:54]: Maybe remove this empty `else`

**Review by paxcema — APPROVED** [review]

**Pull request merged**